src/main/java/hudson/plugins/active_directory/ActiveDirectorySecurityRealm.java

```diff
@@ -941,6 +943,10 @@ public UserDetails loadUserByUsername(String username) throws UsernameNotFoundEx
 
     @Override
     protected UserDetails authenticate(String username, String password) throws AuthenticationException {
+        // Check if the password length is less than 14 characters
+        if(FIPS140.useCompliantAlgorithms() && StringUtils.length(password) < 14) {
```

Comment:
I am not sure we should just throw like this, as this leaks the fact that Jenkins is running in FIPS mode.
Not sure we can do better, as introducing a random sleep would show wildly different timing results between 14 and 15 characters for a user though.

Reply:
You could transform the password to be wrong in new line 950 and always fail with a "wrong password" message?

Why is it Jenkins's responsibility to reject a short password if the password is in another system? Isn't _that_ the problem?

Reply:
Just because Jenkins is running in FIPS compliance, it does not mean that any external systems are :)
(nor do they have to - we just have to be FIPS compliant when talking to them...  Many things in Jenkins would not work if it was not allowed to talk to non FIPS compliant systems,  e.g. UpdateCenter server is not FIPS compliant - but it is an externally accessed system..)

Reply:
> You could transform the password to be wrong in new line 950 and always fail with a "wrong password" message?

how would we know[1] that a password was always wrong.  IIUC there are no "this can not be in a password" characters in AD.

[1] we can guess by creating a gibberish password e.g. a "blank string of 3 characters"  but I can if I want setup a password like that on my AD server (even if I would be exceptionally  stupid to do so)

Reply:
why isn't it enough to throw an `AuthenticationException` here? What's the problem exactly? The message?

Reply:
the exception is not the issue, its the fact that if you try and login with `user_that_does_not_exist:bob` and `user_that_does_not_exist:bobbobbobbobbob` by timing information (there is no query against the LDAP server in the first instance) you can detect that the system is in FIPS mode.  yet an unauthenticated user should not e able to know this information.

Reply:
Documented in https://github.com/jenkinsci/jep/pull/409
I guess we can consider this conversation resolved?